app/controllers/check_ins_controller.rb

```diff
@@ -34,27 +34,23 @@ def ensure_marriage
   end
 
   def show_bootstrap_data
-    check_in_need_satisfaction_ratings = @check_in.need_satisfaction_ratings
+    check_in_need_satisfaction_ratings =
+      @check_in.need_satisfaction_ratings.
+        includes(:emotional_need, :user).
+        order('emotional_needs.name')
+
+    user_ratings_of_partner, partner_ratings_of_user =
+      check_in_need_satisfaction_ratings.
+        partition { it.user == current_user }
 
     bootstrap_data = {
       check_in: @check_in.serializer,
-      user_ratings_of_partner:
-        NeedSatisfactionRatingSerializer.new(
-          check_in_need_satisfaction_ratings.
-            where(user: current_user).
-            includes(:emotional_need).
-            order('emotional_needs.name'),
-        ),
+      user_ratings_of_partner:,
     }
 
     bootstrap_data[:partner_ratings_of_user] =
       if @check_in.submitted_by_both_partners?
-        NeedSatisfactionRatingSerializer.new(
-          check_in_need_satisfaction_ratings.
-            where(user: current_user.marriage.decorate.other_partner).
-            includes(:emotional_need).
-            order('emotional_needs.name'),
-        )
+        partner_ratings_of_user
```

Comment:
This is a now a little less efficient in the `else` case, in which case we have extracted `partner_ratings_of_user` unnecessarily, but it might still be a performance win, on net, since we save a query in this `if` case where we do use the `partner_ratings_of_user`. But, otherwise, `prosopite` will complain.